x/protocolpool/keeper/genesis.go

```diff
@@ -27,11 +27,16 @@ func (k Keeper) InitGenesis(ctx context.Context, data *types.GenesisState) error
 	}
 	for _, budget := range data.Budget {
 		// Validate StartTime
-		if budget.StartTime == nil || budget.StartTime.IsZero() {
-			budget.StartTime = &currentTime
+		if budget.LastClaimedAt == nil || budget.LastClaimedAt.IsZero() {
+			budget.LastClaimedAt = &currentTime
 		}
+		// ignore budgets with period <= 0
+		if budget.Period != nil && budget.Period.Seconds() <= 0 {
+			continue
+		}
```

Comment:
We should also ignore whenever `budget.Period == nil` no? Mayyybe:
```suggestion
		if budget.Period == nil || (budget.Period != nil && budget.Period.Seconds() <= 0) {
			continue
		}
```